cloudimg/aws.py

```diff
@@ -854,3 +903,19 @@ def delete(self, metadata):
                 deleted_snapshot_id = self.delete_snapshot(snapshot)
 
         return deleted_image_id, deleted_snapshot_id
+
+    @staticmethod
+    def _get_tagset(tags):
+        """Convert the metadata tags to the proper value expected by AWS.
+
+        Args:
+            tags (list): List of tags to be converted to the AWS format.
```

Comment:
False, it's a dict.

I'm not that keen on writing doc strings for private methods anyway.  You named this with leading `_` to indicate it's private, which is good, but some people won't get the hint.  Leaving out the doc string (when all the public stuff has doc strings) is another hint which might help them realize maybe they're not supposed to use it.  And all docs need to be maintained or else will eventually become incorrect.

I wasn't going to say anything, but the fact that the doc string contains a mistake prompted me to raise this.  At least docs you don't write can never be wrong.  I also don't usually bother pointing out mistakes in comments rather than doc strings.

By the way, as far as I'm aware using type hints here is a valid option and if you use a modern editor it might have highlighted the mistake.  You do have to be careful with the changes in list/dict type hints over the python 3.x series though (I think only plain old `list` and `dict` would be safe, with no generics available).

Reply:
Good catch!

I renamed this method to `_get_tagdict` and got rid of the doc strings. I agree that private method shouldn't include it.

PTAL again